### ISSUE TITLE
Provide existing KMS Key for init keys encryption

### ISSUE THREAD

**Issue by boguszj** [terraform-google-modules/terraform-google-vault#143]
### TL;DR

Currently KMS key is created by the module. It would be nice to have an ability to provide an existing key. That would allow to use BYOK and make key management easier - less keyrings and keys to keep track of in case they are being managed manually.

### Detailed design

```markdown
# Proposition - not tested yet

locals {
  kms_keyring = {
    id         = "projects/${var.project_id}/locations/${coalesce(var.existing_kms_keyring_location, var.region)}/keyRings/${coalesce(var.existing_kms_keyring, var.kms_keyring)}"
    name       = coalesce(var.existing_kms_keyring, var.kms_keyring)
    location   = coalesce(var.existing_kms_keyring_location, var.region)
    project_id = coalesce(var.existing_kms_keyring_project, var.project_id)
  }

  kms_crypto_key = {
    id               = "projects/${var.project_id}/locations/${coalesce(var.existing_kms_keyring_location, var.region)}/keyRings/${coalesce(var.existing_kms_keyring, var.kms_keyring)}/cryptoKeys/${coalesce(var.existing_kms_crypto_key, var.kms_crypto_key)}"
    name             = coalesce(var.existing_kms_crypto_key, var.kms_crypto_key)
    protection_level = upper(var.kms_protection_level)
  }
}

# Create the KMS key ring if existing_kms_keyring was not provided
resource "google_kms_key_ring" "vault" {
  name     = local.kms_keyring.name
  location = local.kms_keyring.location
  project  = local.kms_keyring.project_id
  count    = var.existing_kms_keyring == null ? 1 : 0
}

# Create the crypto key for encrypting init keys if existing_kms_key was not provided
resource "google_kms_crypto_key" "vault-init" {
  name            = local.kms_crypto_key.name
  key_ring        = local.kms_keyring.id
  rotation_period = "604800s"
  count           = var.existing_kms_crypto_key == null ? 1 : 0

  version_template {
    algorithm        = "GOOGLE_SYMMETRIC_ENCRYPTION"
    protection_level = local.kms_crypto_key.protection_level
  }

  depends_on = [
    google_kms_key_ring.vault
  ]
}
```

### Additional information

Proposed design would create the need to manually depend on `google_kms_crypto_key.vault-init` in resources where now dependency is provided by terraform based on resource usage to make sure that before KMS Key Id is used either:
- one was already created,
- external key was provided 

It seems that manual dependency (`depends_on` block) would be required in:
- `template_file.vault-config`,
- `google_kms_crypto_key_iam_member.ck-iam`,
- `google_kms_secret_ciphertext.vault-tls-key-encrypted`
